### PR TITLE
Add 3DS tests

### PR DESCRIPTION
**Pull request by bg-stripe**
r? @bdorfman-stripe 

I've also updated the `payment_method_not_available` error logic.

### CODE REVIEW

File: Stripe/STPPaymentContext.m [hunk 1]
```diff
@@ -598,15 +598,14 @@ three_d_secure types of required or optional (not_supported are filtered earlier
...
                                                                               returnURL:self.configuration.returnURLBlock().absoluteString
                                                                                    card:cardSource.stripeID];
+    threeDSecureParams.metadata = self.sourceInformation.metadata;
 
     STPSourceCompletionBlock threeDSSourceCompletion = ^(STPSource * _Nullable threeDSSource, NSError * _Nullable error) {
         if (error
             || threeDSSource == nil
             || threeDSSource.flow != STPSourceFlowRedirect) {
             if (!threeDSecureRequired
-                && error.domain == StripeDomain
```
Comment (bdorfman-stripe) [inline Stripe/STPPaymentContext.m]: I think you should still check that the error is a StripeDomain here?

File: Stripe/STPPaymentContext.m [hunk 2]
```diff
@@ -598,15 +598,14 @@ three_d_secure types of required or optional (not_supported are filtered earlier
...
-                && error.code == STPCardError
-                && [error.userInfo[STPCardErrorCodeKey] isEqualToString:STPPaymentMethodNotAvailable]) {
+                && [error.userInfo[STPStripeErrorCodeKey] isEqualToString:@"payment_method_not_available"]) {
```
Comment (bdorfman-stripe) [inline Stripe/STPPaymentContext.m]: I think all the codes should still be converted to constants. I'm not a huge fan of just checking for the raw string values. Is this not still a card error? (given that 3ds sources are cards). At the least we should define string constants for the errors we check for I think.

File: Stripe/STPPaymentContext.m (hunk 1 above)
Comment (bg-stripe) [inline Stripe/STPPaymentContext.m]: 👍

File: Stripe/STPPaymentContext.m (hunk 2 above)
Comment (bg-stripe) [inline Stripe/STPPaymentContext.m]: That's fair. Source errors have type `invalid_request`, not `card_error`, so I'm a bit unhappy with putting this under `STPCardErrorCodeKey`, but I think it's probably easier to just do that.

Comment (bdorfman-stripe) [inline Stripe/STPPaymentContext.m]: I'm fine with not putting it under card_error I guess. If you want to just define an NSString const that is that expected error though? Not 100% sure of the Swift bridging implications of it being in that enumeration or not though.

**Comment by bg-stripe** [comment]
updated!

**Comment by bdorfman-stripe** [comment]
👍

**Pull request merged**